packages/core-modules/test/modules/UpgradeModule.test.js

```diff
@@ -5,50 +5,44 @@ const assertRevert = require('@synthetixio/core-js/utils/assert-revert');
 const { ethers } = hre;
 
 describe('UpgradeModule', () => {
-  let UpgradeModuleMockFactory, UpgradeModuleMock;
+  let UpgradeModule, Implementation;
   let owner, user;
 
   before('identify signers', async () => {
     [owner, user] = await ethers.getSigners();
   });
 
-  before('initialize module', async () => {
-    UpgradeModuleMockFactory = await ethers.getContractFactory('UpgradeModuleMock');
-    UpgradeModuleMock = await UpgradeModuleMockFactory.deploy();
+  before('deploy the implementation', async () => {
+    const factory = await ethers.getContractFactory('ImplementationMock');
+    Implementation = await factory.deploy();
   });
 
-  before('set the owner', async () => {
-    await UpgradeModuleMock.__setOwner(owner.address).then((tx) => tx.wait());
+  before('deploy the module mocking the first owner', async () => {
+    const factory = await ethers.getContractFactory('UpgradeModuleMock');
+    UpgradeModule = await factory.deploy();
+
+    const tx = await UpgradeModule.mockFirstOwner(owner.address);
+    await tx.wait();
+  });
+
+  it('shows that the implementation is not set', async () => {
+    assert.equal(await UpgradeModule.getImplementation(), '0x0000000000000000000000000000000000000000');
   });
 
-  describe('when a regular user attempts to upgrade the system', () => {
+  describe('when attempting to set the implementation with a non owner signer', () => {
```

Comment:
Since all UpgradeModule adds to UUPSProxy is the onlyOwner protection, that's all we test here.

We assume that the rest of the functionality is tested at a lower dependency level.